notebooks/02-ingestion/02-batch-ingestion.ipynb

```diff
@@ -0,0 +1,1008 @@
+    "SELECT\n",
+    "  TIME_PARSE(\"timestamp\") AS \"__time\",\n",
+    "  *\n",
+    "FROM \"ext\"\n",
+    "\n",
+    "WHERE \"isRobot\"='false'\n",
+    "\n",
+    "PARTITIONED BY DAY\n",
+    "'''\n",
+    "\n",
+    "druid.display.run_task(sql)\n",
+    "druid.sql.wait_until_ready('wikipedia_events_only_human')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5c825d39-0bbd-4cd3-ae8a-760146d9fff7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "druid.display.sql(\"\"\"\n",
+    "SELECT isRobot, channel, count(*) num_events\n",
+    "FROM \"wikipedia_events_only_human\" \n",
+    "GROUP BY 1,2 \n",
+    "ORDER BY 3 DESC \n",
+    "LIMIT 10\n",
+    "\"\"\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6561d663-9cfe-4a62-80cc-dcbb0ce199ff",
+   "metadata": {},
+   "source": [
+    "#### Transform Data During Ingestion\n",
+    "\n",
+    "The SQL language provides a rich [set of functions](https://druid.apache.org/docs/latest/querying/sql-scalar.html) that can be applied to input columns to transform the data as it is being ingested. All scalar SQL function are available for normal ingestion. Rollup ingestion is discussed in the [Rollup Notebook](05-rollup.ipynb) which includes the use of aggregate functions at ingestion time as well.\n",
+    "\n",
+    "Here are some examples of such transformations:\n",
+    "\n",
+    "##### Time manipulation\n",
+    "There are many [time parsing and manipulation functions](https://druid.apache.org/docs/latest/querying/sql-scalar.html#date-and-time-functions) available in Apache Druid. It is common to do some time cleansing/transformation at ingestion. Here are some examples of time manipulation functions:\n",
+    "```\n",
+    "  TIME_PARSE( \"timestamp\") AS \"__time\",   \n",
+    "  TIME_FLOOR( TIME_PARSE( \"timestamp\"), 'P1W') AS \"week_start\",\n",
+    "  TIMESTAMPDIFF( DAY,\n",
+    "                 TIME_FLOOR( TIME_PARSE( \"timestamp\"), 'P1W'),\n",
+    "                 TIME_PARSE( \"timestamp\")\n",
+    "               ) AS \"days_since_week_start\"\n",
+    "   \n",
+    "```\n",
+    "\n",
+    "##### Use CASE statements to transform data\n",
+    "CASE statements can be used to resolve complex logic and prepare columns for certain query patterns. \n",
+    "Examples:\n",
+    "```\n",
+    "  CASE\n",
+    "     WHEN UPPER(\"adblock_list\")='NOADBLOCK' THEN 0\n",
+    "     ELSE 1\n",
+    "  END AS adblock_count,\n",
+    "\n",
+    "  CASE\n",
+    "     WHEN UPPER(\"adblock_list\")='EASYLIST' THEN 1\n",
+    "     ELSE 0\n",
+    "  END AS easylist_count\n",
+    "```\n",
+    "The two case statements above are examples of converting a categorical column like `adblock_list` into a numerical column that can be used as a meaningful metric when aggregated across different dimensions to get the count of events that were affected by an ad blocker.\n",
+    "\n",
+    "##### String manipulation\n",
```

Comment:
Add link to the string functions notebook here

Reply:
see comment above